pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -205,9 +205,10 @@ private static void createLeadControllerResourceIfNeeded(String helixClusterName
     if (resourceConfig == null) {
       resourceConfig = new ResourceConfig(LEAD_CONTROLLER_RESOURCE_NAME);
     }
-    // Set RESOURCE_ENABLED to false if it's absent in resource config
-    if (resourceConfig.getSimpleConfig(LEAD_CONTROLLER_RESOURCE_ENABLED_KEY) == null) {
-      resourceConfig.putSimpleConfig(LEAD_CONTROLLER_RESOURCE_ENABLED_KEY, Boolean.FALSE.toString());
+    // Explicitly set RESOURCE_ENABLED to true
+    String leadControllerResourceEnabled = resourceConfig.getSimpleConfig(LEAD_CONTROLLER_RESOURCE_ENABLED_KEY);
+    if (!Boolean.parseBoolean(leadControllerResourceEnabled)) {
```

Comment:
We should only set this flag to true if it is not explicitly configured. The current logic will ignore the current config and always override the config

Reply:
Yes, we should always override the config to true. The `if` statement above checks whether the config is empty or false; if that's the case, we set it to true. If the config is already true, do nothing. 
The null check has been handled in `Boolean.parseBoolean`, thus merging the logic to one place.

Reply:
Discussed offline. We want to enable this feature in all the clusters to keep the behavior consistent in all cluster. Thus, we enable the feature in this PR. In the next official release, we'll remove the config to always use this lead controller resource feature.